<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="x_init" default="0" />
  <arg name="y_init" default="0" />
  <arg name="z_init" default="1" />
  <arg name="VEHICLE_NAME" default="bicopter_through"/>

  <!-- PID parameters -->
  <rosparam file="$(find flight_controller)/config/pid_bicopter_through.yaml"/>


  
  <!-- Joystick -->
  <node name="joy" pkg="joy" type="joy_node" output="screen" respawn="true">
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <!-- Controller -->
  <node name="flight_controller" pkg="flight_controller" type="bicopter_through" respawn="true" output="screen">
    <param name="QUAD_NAME" value="$(arg VEHICLE_NAME)"/>
    <param name="X_INIT" value="$(arg x_init)"/>
    <param name="Y_INIT" value="$(arg y_init)"/>
    <param name="Z_INIT" value="$(arg z_init)"/>
  </node>
  
  <!-- SBUS communication -->
  <node name="serial_to_sbus" pkg="serial_to_sbus" type="serial_to_sbus" respawn="true" output="screen">
    <param name="USB" value="/dev/ttyUSB0"/>
  </node>

  <node pkg="flight_controller" type="uav_kf.py" name="uavKf" output="screen">
    <param name="device_name" value="$(arg VEHICLE_NAME)"/>
  </node>

  <!-- VRPN -->
  <arg name="server" default="192.168.3.3"/>
  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg server)
      port: 3883
      update_frequency: 100.0
      frame_id: world
      use_server_time: false
      broadcast_tf: false
      refresh_tracker_frequency: 100.0
    </rosparam>
  </node>

  <!-- Bagging the data -->
  <!-- <node pkg="rosbag" type="record" name="bag_record" args="-a -o /home/xianghe/LOG_AND_BAG/output.bag"/> -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find flight_controller)/rviz/through_rviz.rviz" required="true" />
</launch>
